week3/solutions/hangman/hangman.py

```diff
@@ -0,0 +1,122 @@
+import os
+import random
+import string
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+WORDLIST_FILENAME = os.path.join(current_dir, 'words.txt')
+
+def load_words():
+  """
+  Returns a list of valid words. Words are strings of lowercase letters.
+    
+  Depending on the size of the word list, this function may
+  take a while to finish.
+  """
+  print("Loading word list from file...")
+  in_file = open(WORDLIST_FILENAME, 'r')
+  line = in_file.readline()
+  wordlist = line.split()
+  print("  ", len(wordlist), "words loaded.")
+  return wordlist
```

Comment:
In Python we use a "context generator" using the `with` statement:

This code specifically reads a one-line file. This could blow up in memory depending on that size. You can write this simpler as:
```
def load_words():
	with open(WORDLIST_FILENAME, 'r') as in_file:
		return sum([len(cur_line) for cur_line in in_file])
```